routemaster/state_machine/utils.py

```diff
@@ -151,18 +174,15 @@ def labels_to_retry_for_action(
 
     active_participants = select((
         ranked_transitions.c.label_name,
-        labels.c.metadata,
     )).where(and_(
-        ranked_transitions.c.new_state == action.name,
+        ranked_transitions.c.new_state == gate.name,
```

Comment:
Is there a secondary helper here we can pull out? Something like `_labels_in_state(state_machine, state, conn) -> QueryExpression`?

I'm assuming that this would remove the duplication of this helper with the very similar `labels_in_state` above and we'd just onwards filter the query expression onwards with `not(metadata_triggers_processed)` here. Not sure if that's possible with sqlalchemy though.

Reply:
bcd329c8313f32137281a598e71b283391117d0e